JustSaying.UnitTests/AwsTools/MessageHandling/Sns/TopicByName/WhenPublishingAsync.cs

```diff
@@ -42,19 +43,21 @@ protected override Task Given()
 
         protected override async Task When()
         {
-            await SystemUnderTest.PublishAsync(new SimpleMessage
+            var publishData = new PublishEnvelope(new SimpleMessage())
             {
                 MessageAttributes = new Dictionary<string, Models.MessageAttributeValue>
                 {
                     {
-                        MessageAttributeKey,
-                            new Models.MessageAttributeValue{
+                        MessageAttributeKey, new Models.MessageAttributeValue
+                            {
                                 StringValue = MessageAttributeValue,
                                 DataType = MessageAttributeDataType
                             }
                     }
                 }
-            });
+            };
+
+            await SystemUnderTest.PublishAsync(publishData, CancellationToken.None);
```

Comment:
An overload (extension method) that accepts the message itself and wraps it in an envelope for you would be nice.